toolchain/check/eval.cpp

```diff
@@ -781,21 +782,21 @@ static auto ReplaceAllFieldsWithConstantValues(EvalContext& eval_context,
 
 // Given an instruction and its ID, replaces its type with the corresponding
 // value in this evaluation context. Updates `*phase` to describe the phase of
-// the result, and returns whether `*phase` is a constant phase.
+// the result, and returns whether `*phase` is a non-error constant phase.
 static auto ReplaceTypeWithConstantValue(EvalContext& eval_context,
```

Comment:
The callers of this generally look like `ReplaceTypeWithConstantValue(...) && ReplaceAllFieldsWithConstantValues(...)` so we'll still inconsistently produce an error constant if the type is an error but one of the operands is runtime.

Reply:
Oh I see what you're saying, yeah. Thanks. Maybe the short circuits do all need to adjust then. I will take another crack at this.